drivers/i2s/i2s_ll_stm32.c

```diff
@@ -893,6 +931,107 @@ static const struct device *get_dev_from_tx_dma_channel(uint32_t dma_channel)
 	.mem_block_queue.len = ARRAY_SIZE(dir##_##index##_ring_buf)	\
 }
 
+/*****************************************************************************/
```

Comment:
This is globally an interesting addition, though, it is not I2S specific and should rather be moved to pinctrl driver.
This would better be done as a stand alone PR though. 
@gmarull is that ok for you ?
I know this is rather STM32 (/STM32-like) specific for now, but could maybe be extended to suit other vendors.

Reply:
Glad you picked up on this :)

The implementation is sufficiently generic to cater for the STM32 family, originally I thought about moving into the HAL STM32 module but couldn't find a suitable location. Perhaps it could be extended further to other vendors by specifying `PIN` and `PORT_RANGE` macros accordingly.

I would be happy to move it over to a better place, breaking it into its own PR as result, making it more reusable that way; as long as I can be provided with some guidance on the whereabouts.

Reply:
> as long as I can be provided with some guidance on the whereabouts.

This is where i hope @gmarull can help us